### PR TITLE
Create compile/mark/init for initializing orient for MarkDef + Remove initTextConfig()

### PR DESCRIPTION
**Pull request by kanitw**
Please merge #1967 first!

### CODE REVIEW

File: src/channel.ts [hunk 1]
```diff
@@ -120,7 +120,7 @@ export function getSupportedMark(channel: Channel): SupportedMark {
     case SIZE:
       return {
         point: true, tick: true, rule: true, circle: true, square: true,
-        bar: true, text: true
+        bar: true, text: true, line: true
```
Comment (domoritz) [inline src/channel.ts]: good catch

File: src/compile/common.ts [hunk 2]
```diff
@@ -44,15 +44,21 @@ export function applyConfig(e: VgEncodeEntry,
   return e;
 }
 
-export function applyMarkConfig(e: VgEncodeEntry, model: UnitModel, propsList: string[]) {
-  return applyConfig(e, model.config.mark, propsList);
+export function applyMarkConfig(e: VgEncodeEntry, model: UnitModel, propsList: (keyof MarkConfig)[]) {
+  propsList.forEach(function(property) {
```
Comment (domoritz) [inline src/compile/common.ts]: Why not `property => {}`

File: src/compile/common.ts [hunk 3]
```diff
@@ -44,15 +44,21 @@ export function applyConfig(e: VgEncodeEntry,
...
+  });
+  return e;
 }
 
 /**
  * Return value mark specific config property if exists.
  * Otherwise, return general mark specific config.
  */
-export function getMarkConfig(prop: keyof MarkConfig, mark: Mark, config: Config) {
+export function getMarkConfig<P extends keyof MarkConfig>(prop: P, mark: Mark, config: Config): MarkConfig[P] {
```
Comment (domoritz) [inline src/compile/common.ts]: nice!

File: src/compile/mark/init.ts [hunk 4]
```diff
@@ -1,60 +1,57 @@
...
+import {Dict, contains} from '../../util';
+import {Scale, hasDiscreteDomain} from '../../scale';
+import {isFieldDef, isMeasure, FieldDef} from '../../fielddef';
+import {TEMPORAL} from '../../type';
+import {Config} from '../../config';
+import {getMarkConfig} from '../common';
+import {StackProperties} from '../../stack';
+
+export function initMarkDef(mark: Mark | MarkDef, encoding: Encoding, scale: Dict<Scale>, config: Config): MarkDef & {filled: boolean} {
+  let markDef = isMarkDef(mark) ? mark : {type: mark};
```
Comment (domoritz) [inline src/compile/mark/init.ts]: const

File: src/compile/mark/rule.ts [hunk 5]
```diff
@@ -12,7 +12,7 @@ export const rule: MarkCompiler = {
   role: undefined,
   encodeEntry: (model: UnitModel) => {
     let e: VgEncodeEntry = {};
```
Comment (domoritz) [inline src/compile/mark/rule.ts]: const?

File: src/compile/mark/text.ts [hunk 6]
```diff
@@ -76,3 +80,11 @@ function textRef(textDef: TextFieldDef | ValueDef<any>, config: Config): VgValue
   }
   return {value: config.text.text};
 }
+
+function align(encoding: Encoding, config: Config) {
+  const alignConfig = getMarkConfig('align', 'text', config);
+  if (alignConfig === undefined) {
+    return channelHasField(encoding, X) ? 'center' : 'right';
+  }
+  return undefined;
```
Comment (domoritz) [inline src/compile/mark/text.ts]: why return undefined here?

File: src/compile/model.ts [hunk 7]
```diff
@@ -239,12 +239,18 @@ export abstract class Model {
 
   protected abstract getMapping(): {[key: string]: any};
 
-  public reduce<T, U>(f: (acc: U, fd: FieldDef, c: Channel) => U, init: T, t?: any) {
-    return reduce(this.getMapping(), f, init, t);
+  public reduceFieldDef<T, U>(f: (acc: U, fd: FieldDef, c: Channel) => U, init: T, t?: any) {
+    return reduce(this.getMapping(), (acc:U , cd: ChannelDef, c: Channel) => {
+     return isFieldDef(cd) ? f(acc, cd, c) : acc;
```
Comment (domoritz) [inline src/compile/model.ts]: indentation is odd

File: src/mark.ts [hunk 8]
```diff
@@ -242,10 +240,84 @@ export interface MarkConfig {
...
+  /**
+   * The font weight (e.g., `"normal"`, `"bold"`, `900`).
+   */
+  fontWeight?: FontWeight | FontWeightNumber;
+
 }
 
 export const defaultMarkConfig: MarkConfig = {
-  color: '#4682b4',
+  color: '#4c78a8',
```
Comment (domoritz) [inline src/mark.ts]: Why?

File: src/compile/mark/rule.ts (hunk 5 above)
Comment (kanitw) [inline src/compile/mark/rule.ts]: doesn't matter This will be removed in a follow-up PR.

File: src/compile/mark/text.ts (hunk 6 above)
Comment (kanitw) [inline src/compile/mark/text.ts]: No need to inject value.  Vega deals with config for us.

File: src/compile/common.ts [hunk 9]
```diff
@@ -44,15 +44,21 @@ export function applyConfig(e: VgEncodeEntry,
   return e;
 }
 
-export function applyMarkConfig(e: VgEncodeEntry, model: UnitModel, propsList: string[]) {
-  return applyConfig(e, model.config.mark, propsList);
+export function applyMarkConfig(e: VgEncodeEntry, model: UnitModel, propsList: (keyof MarkConfig)[]) {
+  propsList.forEach((property) => {
```
Comment (domoritz) [inline src/compile/common.ts]: No need to put () around a single parameter

File: src/compile/mark/rule.ts (hunk 5 above)
Comment (domoritz) [inline src/compile/mark/rule.ts]: K

File: src/compile/common.ts (hunk 9 above)
Comment (domoritz) [inline src/compile/common.ts]: But it doesn't really matter either.

**Pull request merged**